src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsRoleEnvironmentTelmetryInitializer.cs

```diff
@@ -32,34 +35,27 @@ public void Initialize(ITelemetry telemetry)
                 return;
             }
 
-            // We cannot cache these values as the environment variables can change on the fly.
-            if (string.IsNullOrEmpty(telemetry.Context.Cloud.RoleName))
+            Lazy<string> siteSlotName = new Lazy<string>(() =>
             {
-                telemetry.Context.Cloud.RoleName = GetRoleName();
-            }
+                // We cannot cache these values as the environment variables can change on the fly.
+                return GetAzureWebsiteUniqueSlotName();
```

Comment:
Previously we were looking up env values and formatting the site name multiple times for each log.